overlord/ifacestate/ifacemgr.go

```diff
@@ -64,26 +64,35 @@ func Manager(s *state.State) (*InterfaceManager, error) {
 // Connect returns a set of tasks for connecting an interface.
 //
 func Connect(s *state.State, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
-	// TODO: Store the intent-to-connect in the state so that we automatically
-	// try to reconnect on reboot (reconnection can fail or can connect with
-	// different parameters so we cannot store the actual connection details).
+	plugRef := interfaces.PlugRef{Snap: plugSnap, Name: plugName}
+	slotRef := interfaces.SlotRef{Snap: slotSnap, Name: slotName}
+	// Store the intent to connect
+	intents := getIntents(s)
+	intents.Add(Intent{Plug: plugRef, Slot: slotRef})
+	setIntents(s, intents)
```

Comment:
how will you handle when you have the intent, and a pending change for the same connection because of a restart? shouldn't intents be just called plug-connections and be set when the change is about done?